---
title: "whoami: a brief introduction"
date: 2023-11-15T11:25:12+05:30
---



The _whoami_ command amused me when I first learned it. Here I was, asking my shell one of the most profound questions in human history:

```shell
$ whoami                            
```

My shell, oblivious to any such philosophical musings, always echoes the same answer back:

```bash
ira
```

Welcome to my blog! My name is Ira Zibbu. I spend most of my time fiddling around computers and thinking about evolution. I am currently finishing my Master's in Science at the Indian Institute of Science Education and Research, Thiruvananthapuram. I'm broadly interested in experimental evolution, evolutionary dynamics, microbial genomics and genetics, next-generation sequencing, and synthetic biology.

Physicists have spent a lot of time looking for a grand theory to unify the discipline; biologists were gifted one with the theory of evolution in the 19th century. I don't think evolution alone has the explanatory power to describe all biological processes, but rather that evolution is the framework within which all of biology is contextualised. As an undergraduate, I was introduced to evolution and genetics as disparate fields
In the former, the evolutionary events are explained, but the underlying molecular mechanisms are usually not investigated. In the latter, comprehensive mechanistic descriptions of biological phenomenon are uncovered, but little thought is given to the bigger picture of how these processes evolved. Fortunately, Julian Huxley (and many others) had already though about this before me, and Modern Synthesis was born 60 years before I was. We also figured out the really neat trick of sequencing nucleic acids, and blew the doors wide open for molecular evolution to grow. I was born with the proverbial silver spoon of next-generation sequencing in my mouth and have always had a deep appreciation for its power. The third of Clarke's three laws states that any sufficiently advanced technology is indistinguishable from magic, and you cannot convince me that sequencing technologies are anything short of molecular wizardry. 

This, I suppose, is where my research comes in. I'm drawn to abstract questions about the nature of evolution, and I fell in love with experimental evolution for its ability to answer some of these questions. Is evolution an idiosyncratic or a highly repeatable process? Can fitness increase indefinitely? How is phenotypic evolution coupled with molecular evolution? Because evolution experiments give us direct access to the intermediate transitional forms generated as a population evolves, we can essentially build a frame-by-frame 'movie' of evolution. It's also open-ended, and tends to generate very information-rich data that can be repeatedly analysed in different ways. The inherent 'creativity' of the evolutionary process means evolution experiments are gifts that keep on giving.

Right now, I work with the Lenski Long-Term Evolution Experiment, where I am trying to understand what kinds of chromosomal rearrangements have happened over 75,000 generations of the experiment. I also tinkered around with some machine learning and proteins, chemotaxis and PDEs and insect ecology. 

Here is some other stuff I like:

- Microbes: They were here before us, and will be there after us. They account for the vast majority of the tree of life and exhibit incredible physiological, ecological and sequence diversity. I like lab staples like *E. coli* for how easy they are to handle and manipulate. They're great for when you are asking tough questions about evolution, and need a well-understood model to work with. I also like archeae, because these microbes really push the boundaries of what is possible in a biological system. 
  
- Synthetic biology: The Lego bricks of biology! If watching cells do neat stuff was great, then programming them to do it is even better. If you'd like, you can throw in some ideas from evolution in the mix and get directed evolution experiments. 

- Linux and FOSS: I made the switch to Linux about a year ago, so I am still at the tip of the iceberg with Ubuntu, but UNIX based systems just blew my mind. These days most of my work is bioinformatics, so switching to a UNIX-based OS was inevitable, but I can't believe how much I missed out on, solely because \[redacted] company has a monopoly over the default OS on most computers sold today. As someone put it, with Linux, you can build castles in the clouds or you can repeatedly shoot yourself in the foot. I tend to do a lot of the latter*

I also mentor many undergraduate students with navigating science, academia and finding opportunities. If you have a question, or an application you want help with, drop me an email!

If you would like to get in touch for other stuff, please feel free to email me at ira.zibbu@gmail.com. I like talking to new people.

---

\* I once wanted to delete every subdirectory in a directory, but instead of running `rm -rf */` I ran `rm -rf /*` 

